src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -19,22 +26,183 @@ const ActorInit Obj_Boyo_InitVars = {
+    (ActorFunc)ObjBoyo_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Init.s")
+typedef void (*ObjBoyoCollisionHandler)(ObjBoyo* this, Actor* actor);
+
+typedef struct ObjBoyoUnkStruct {
+    /* 0x0 */ s16 id;
+    /* 0x4 */ ObjBoyoCollisionHandler colHandler;
+} ObjBoyoUnkStruct; // size = 0x8
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_TYPE_PLAYER | AC_HARD | AC_ON,
+        OC1_TYPE_2 | OC1_TYPE_1 | OC1_TYPE_PLAYER | OC1_ON,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x01CBFFBE, 0x00, 0x00 },
+        TOUCH_NONE,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    {
+        60,
+        140,
+        0,
+        { 0, 0, 0 },
+    },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static ObjBoyoUnkStruct sCollisionHandlers[] = {
+    { ACTOR_PLAYER, ObjBoyo_PushPlayer },
+    { ACTOR_EN_KAIZOKU, ObjBoyo_PushPirate },
+    { ACTOR_EN_BOM, ObjBoyo_ExplodeBomb },
+};
+
+void ObjBoyo_Init(Actor* thisx, PlayState* play) {
+    ObjBoyo* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->animatedMaterial = Lib_SegmentedToVirtual(&object_boyo_Matanimheader_000E88);
+}
+
+void ObjBoyo_Destroy(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void ObjBoyo_PushPlayer(ObjBoyo* this, Actor* actor) {
+    Player* player = (Player*)actor;
+
+    player->pushedSpeed = 30.0f;
+    player->pushedYaw = this->actor.yawTowardsPlayer;
+}
+
+void ObjBoyo_PushPirate(ObjBoyo* this, Actor* actor) {
+    EnKaizoku* kaizoku = (EnKaizoku*)actor;
+
+    kaizoku->unk_2F0 = 30.0f;
+    kaizoku->unk_2F4 = Actor_WorldYawTowardActor(&this->actor, &kaizoku->picto.actor);
+}
+
+void ObjBoyo_ExplodeBomb(ObjBoyo* this, Actor* actor) {
+    EnBom* bom = (EnBom*)actor;
+
+    bom->timer = 0;
+}
+
+/*
+ * Find the Actor we collided with.
+ */
+Actor* ObjBoyo_FindCollidedActor(ObjBoyo* this, PlayState* play, s32* index) {
+    Actor* collidedActor;
+    ObjBoyoUnkStruct* iter;
+    s32 i;
+
+    if (this->collider.base.ocFlags2 & OC2_HIT_PLAYER) {
+        *index = 0;
+        return &GET_PLAYER(play)->actor;
+    }
+    iter = sCollisionHandlers + 1;
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        collidedActor = this->collider.base.oc;
+        for (i = 1; i != 3; i++) {
```

Comment:
```suggestion
        for (i = 1; i < 3; i++) {
```
Does this still match?

Reply:
bump. You mentioned it matches but maybe you forgot to change it?